recipes/c-blosc/all/conanfile.py

```diff
@@ -77,58 +74,60 @@ def requirements(self):
             self.requires("zstd/1.5.2")
 
     def source(self):
-        tools.files.get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BLOSC_INSTALL"] = True
+        tc.variables["BUILD_STATIC"] = not self.options.shared
+        tc.variables["BUILD_SHARED"] = self.options.shared
```

Comment:
So close to the right one yet so far https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html